RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,248 @@
+
+
+## Execution Phase
+
+The following section describes how both parties should interact with the Bitcoin blockchain during the [RFC003 execution phase](./RFC-003-SWAP-basic.md#execution-phase).
+
+### Deployment
+
+At the start of the deployment stage, both parties compile the contract into bytes as described in the previous section.
+We will call this value `contract_script`.
+
+To deploy the Bitcoin HTLC, the *funder* must confirm a transaction on the relevant Bitcoin blockchain.
+One of the transaction's outputs must have the following properties:
+
+- Its `value` MUST be equal to the `quantity` parameter in the Bitcoin asset header.
+- It MUST have a Pay-To-Witness-Script-Hash (P2WSH) `scriptPubKey` derived from `contract_script` (See [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#specification) for how to construct the `scriptPubkey` from the `contract_script`).
+
+To be notified of the deployment event, both parties MAY watch the blockchain for a transaction with an output matching the required `scriptPubkey` and having the required value.
+
+### Redeem
+
+Before redeeming, *the redeemer* SHOULD wait until the deployment transaction is included in the Bitcoin blockchain with enough confirmations such that they consider it permanent.
+
+To redeem the HTLC, the redeemer MUST submit a transaction to the blockchain which spends the P2WSH output.
+The redeemer can use following witness data to spend the output if they know the `secret`:
+
+| Data             | Description                                                                                             |
+|:-----------------|:--------------------------------------------------------------------------------------------------------|
+| redeem_signature | A valid SECP256k1 ECDSA DER encoded signature on the transaction with respect to the `redeem_pubkey`    |
+| redeem_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `redeem_identity` |
+| secret           | The pre-image of the `secret_hash` under the `hash_function`                                            |
+| `01`             | A single byte used to activate the redeem path in the `OP_IF`                                           |
+| contract_script  | The compiled contract (as generally required when redeeming from a P2WSH output)                        |
+
+For how to use this witness data to construct the redeem transaction see [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#transaction-id).
+
+To be notified of the redeem event, both parties MAY watch the blockchain for transactions that spend from the output and check that the witness data is in the above form.
+If Bitcoin is the `beta_ledger`, then the funder (Bob) MUST watch for such a transaction and then extract the `secret` from the witness data and continue the protocol.
```

Comment:
:upside_down_face: Same as for Ethereum, this is the first time that `beta_ledger` and Bob are mentioned in this document. I would link to the [relevant section](https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-basic.md#4-bob-redeems) in RFC003. Could also extract it into its own subsection.

Reply:
I linked to it but it sucks because you can't really figure out what is meant from it by looking at RFC003. I made a comment here: https://github.com/comit-network/RFCs/issues/19